### DISCUSSION TITLE
Postgres XMIN value

### DISCUSSION THREAD

**Discussion by louds1lence** [diesel-rs/diesel#3051] [Q&A]
Recently I found myself wanting to use XMIN values for soft row-locking capabilities on entities. Documented here: https://www.postgresql.org/docs/14/ddl-system-columns.html

The way I understand it, the main issue here is that in Diesel, PG Integer values are mapped to Rust i32, yet the XMIN is by definition the equivalent of a u32 in Rust. So the way I understand it, it's not currently possible to map a PG value that ends up being a u32 in Rust.

I'm not opposed to doing the work myself and submitting a pull request to make this possible, I'm just looking for input from the project owner/community as to how we should be dealing with this.

**Comment by weiznich** [comment]
Yes postgresql transaction ids are 32 bit width unsigned integers and yes diesel maps postgresqls integer i32 because postgresqls integer are signed. The point is that these types are just distinct types on sql side and so they should be different types on rust side.
How to fix that? As quick fix you could just define the corresponding sql type in your crate using the procedure to define custom types. That allows you to define at least some partial mapping directly to u32 and a complete mapping to a new type wrapper around u32. 
Beside of that we may accept a definition + the corresponding mapping for an postgres specific xid type in diesel if someone opens a PR.
